### PR TITLE
[Fixes #266] Persisting of GeoNode and Celery logs

### LINKED ISSUE #266: GeoNode logs are not persistent in Docker
GeoNode logs are not persistent across container reboots, as the log file is not part of any volume.

https://github.com/GeoNode/geonode-project/blob/7393bb71b14691d4ce135d84fd3c1b1b0c9c2eb5/src/uwsgi.ini#L4

https://github.com/GeoNode/geonode-project/blob/7393bb71b14691d4ce135d84fd3c1b1b0c9c2eb5/docker-compose.yml#L11-L17

### PR DESCRIPTION
**Pull request by jkariscodes**
Edited docker config to allow persistence of geonode [log](https://github.com/GeoNode/geonode-project/blob/7393bb71b14691d4ce135d84fd3c1b1b0c9c2eb5/src/uwsgi.ini#L4) and the celery [log] to a persistent volume called `logs`.

### CODE REVIEW

**Review by randomorder — APPROVED** [review]
Looks good to me, thanks

**Comment by frafra** [comment]
```
GEONODE_LOG_STDOUT=true
CELERY_LOG_STDOUT=true
```

These two variables are defined, but not used.
Wouldn't it better to just have a local switch for logging to STDOUT/STDERR or to the usual file log path?

File: src/entrypoint.sh [hunk 1]
```diff
@@ -9,7 +9,11 @@ invoke () {
     then
         /usr/local/bin/invoke $@
     else
-        /usr/local/bin/invoke $@ >> /usr/src/{{project_name}}/invoke.log 2>&1
+        if [ $IS_CELERY = 'true'] || [ $IS_CELERY = 'True' ]
```
Comment (giohappy) [inline src/entrypoint.sh]: What's the purpose of this new block? 
Each container already has its own local invoke log file `/usr/src/{{project_name}}/invoke.log`, why are you differentiating the name in `/usr/src/geonode_proj/invoke.log` for GeoNode and `/usr/src/{{project_name}}/invoke.log` for Celery?

File: .env.sample [hunk 2]
```diff
@@ -9,6 +9,8 @@ BACKUPS_VOLUME_DRIVER=local
 C_FORCE_ROOT=1
 FORCE_REINIT=false
 INVOKE_LOG_STDOUT=true
+GEONODE_LOG_STDOUT=true
+CELERY_LOG_STDOUT=true
```
Comment (giohappy) [inline .env.sample]: I don't see these new `GEONODE_LOG_STDOUT` and `CELERY_LOG_STDOUT` ever used.
Their purpose should have been to either send logs to a file or redirect to stdout.
For GeoNode this would been tweaking the [uwsgi logger](https://github.com/GeoNode/geonode-project/blob/7393bb71b14691d4ce135d84fd3c1b1b0c9c2eb5/src/uwsgi.ini#L4) 
For Celery it means working on [this line](https://github.com/GeoNode/geonode-project/blob/master/src/celery.sh#L2).

If it's to complicated we can leave this option out for the moment, and just implement the persistent logging into a volume.

File: docker-compose.yml [hunk 3]
```diff
@@ -14,6 +14,7 @@ x-common-django:
     - backup-restore:/backup_restore
     - data:/data
     - tmp:/tmp
+    - logs:/var/log/
```
Comment (giohappy) [inline docker-compose.yml:17]: are we sure that having a shared `/var/log` root folder between multiple containers is fine? This is a standard path for many services and applications, beyond GeoNode and Celery. I fear that might race to write to the same positions. Is my suspect wrong?

Look how many people write inside /var/log, for example  inside the GeoNode container:
image

File: src/entrypoint.sh (hunk 1 above)
Comment (jkariscodes) [reply src/entrypoint.sh]: I deemed them as separate log files and checking whether the log output comes from a celery container or not distunguished by `IS_CELERY` env variable.

Comment (giohappy) [reply src/entrypoint.sh]: these logs are saved locally inside the container (`/usr/src/{{project_name}}/invoke.log`), so we already know which container they refer to. These are distinct from the actual logs, that are sent to `/var/log`.

File: .env.sample (hunk 2 above)
Comment (jkariscodes) [reply .env.sample]: I propose to leave these `GEONODE_LOG_STDOUT` and `CELERY_LOG_STDOUT` options out for the moment.

File: src/entrypoint.sh [hunk 4]
```diff
@@ -9,7 +9,7 @@ invoke () {
     then
         /usr/local/bin/invoke $@
     else
-        /usr/local/bin/invoke $@ >> /usr/src/{{project_name}}/invoke.log 2>&1
+        /usr/local/bin/invoke $@ >> /usr/src/geonode_proj/celery.log 2>&1
```
Comment (afabiani) [inline src/entrypoint.sh]: `geonode_proj` should be reverted to `{{project_name}}`

**Review by afabiani — CHANGES_REQUESTED** [review]

File: src/entrypoint.sh (hunk 4 above)
Comment (jkariscodes) [reply src/entrypoint.sh]: Done

**Review by jkariscodes — COMMENTED** [review]
I have made the changes.

File: src/entrypoint.sh [hunk 5]
```diff
@@ -9,7 +9,7 @@ invoke () {
     then
         /usr/local/bin/invoke $@
     else
-        /usr/local/bin/invoke $@ >> /usr/src/{{project_name}}/invoke.log 2>&1
+        /usr/local/bin/invoke $@ >> /usr/src/{{project_name}}/celery.log 2>&1
```
Comment (giohappy) [inline src/entrypoint.sh:12]: @jkariukidev why did you rename this to `celery.log`? It's the log of the tasks invocation, which is executed both from the `django` and the `celery` containers. 
No need to change things not related to the issue...

File: src/entrypoint.sh (hunk 4 above)
Comment (giohappy) [reply src/entrypoint.sh]: @jkariukidev `celery.log` should be reverted to `invoke.log`. This log is not specific to celery.
No need to touch parts that are not related to the tasks of this issue....

File: docker-compose.yml (hunk 3 above)
Comment (giohappy) [reply docker-compose.yml:17]: I don't know why this comment was marked as resolved but, to me, it isn't resolved. Have you tested the effects of having a shared `/var/log` between multiple containers @jkariukidev?

Comment (jkariscodes) [reply docker-compose.yml:17]: Yes. In this case, different services are writing to the directory at the same time and would be better to distinguish the services present in GeoNode where they write to. I have tested with `/opt` which can be accessed by various services of geonode.
image
and the logs are being written in the following fashion. 
image
